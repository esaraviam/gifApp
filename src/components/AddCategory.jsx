import React ,{ useState }from 'react'

import PropTypes from 'prop-types'
import TextField from '@material-ui/core/TextField'

const AddCategory = ({categories,  setCategories } ) => {
    
    const [newCategory, setNewCategory] = useState("")

    const addCategory = () => {
        if(newCategory.length > 0 ) {
            setCategories(categories => [...categories, newCategory])
            setNewCategory('');
        }
    }
    const handleText = (e) => setNewCategory(e.target.value);

    const handleSubmit = (e) =>{
        e.preventDefault();
        addCategory();
    }

    return (
        <form onSubmit={handleSubmit}>
            <h3>Add Category</h3>
            <TextField 
            fullWidth
            onChange={handleText} 
            label="Please Add the cartegory" 
            value={newCategory}/>
            
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired,
}

export default AddCategory
