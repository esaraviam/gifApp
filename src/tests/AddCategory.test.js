import '@testing-library/jest-dom'
import TextField from '@material-ui/core/TextField'
import AddCategory from '../components/AddCategory'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';

describe('Check Add category component', () => {
    const setCategories = jest.fn()
    let wrapper = shallow( < AddCategory setCategories={setCategories}/ > )

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow( < AddCategory setCategories={setCategories}/ > )
    })


    test('should load Add category properly', () => {
        const snapshot = toJson(wrapper)
        expect(snapshot).toMatchSnapshot();
    })

    test('should input a new category', () => {
       
        const input = wrapper.find('h3')
        console.log(input.html())
        //input.simulate('change' , {target: {value: 'Hola Mundo'}})
    })

    test('Should not post if the input is empty', () => {
        const form = wrapper.find('form')
        form.simulate('submit' , { preventDefault(){}})
        expect(setCategories).not.toHaveBeenCalled()
        
    });
    
    




});