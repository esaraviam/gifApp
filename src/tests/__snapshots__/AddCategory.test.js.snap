// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Add category component should load Add category properly 1`] = `
<form
  onSubmit={[Function]}
>
  <h3>
    Add Category
  </h3>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    label="Please Add the cartegory"
    onChange={[Function]}
    value=""
  />
</form>
`;
